src/QirRuntime/lib/QIR/bridge-qis.ll

```diff
@@ -29,122 +29,264 @@
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__rx(double %.theta, %class.QUBIT* %q)
+  call void @quantum__qis__t__adj(%class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__ry__body(double %.theta, %Qubit* %.q) {
+define void @__quantum__qis__t__ctl(%Array* %.ctls, %Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__ry(double %.theta, %class.QUBIT* %q)
+  %ctls = bitcast %Array* %.ctls to %struct.QirArray*
+  call void @quantum__qis__t__ctl(%struct.QirArray* %ctls, %class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__rz__body(double %.theta, %Qubit* %.q) {
+define void @__quantum__qis__t__ctladj(%Array* %.ctls, %Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__rz(double %.theta, %class.QUBIT* %q)
+  %ctls = bitcast %Array* %.ctls to %struct.QirArray*
+  call void @quantum__qis__t__ctladj(%struct.QirArray* %ctls, %class.QUBIT* %q)
   ret void
 }
 
 define void @__quantum__qis__x__body(%Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__x(%class.QUBIT* %q)
+  call void @quantum__qis__x__body(%class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__y__body(%Qubit* %.q) {
+define void @__quantum__qis__x__ctl(%Array* %.ctls, %Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__y(%class.QUBIT* %q)
+  %ctls = bitcast %Array* %.ctls to %struct.QirArray*
+  call void @quantum__qis__x__ctl(%struct.QirArray* %ctls, %class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__z__body(%Qubit* %.q) {
+define void @__quantum__qis__y__body(%Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
-  call void @quantum__qis__z(%class.QUBIT* %q)
+  call void @quantum__qis__y__body(%class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__crx__body(%Array* %.ctls, double %.theta, %Qubit* %.q) {
+define void @__quantum__qis__y__ctl(%Array* %.ctls, %Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
   %ctls = bitcast %Array* %.ctls to %struct.QirArray*
-  call void @quantum__qis__crx(%struct.QirArray* %ctls, double %.theta, %class.QUBIT* %q)
+  call void @quantum__qis__y__ctl(%struct.QirArray* %ctls, %class.QUBIT* %q)
   ret void
 }
 
-define void @__quantum__qis__crz__body(%Array* %.ctls, double %.theta, %Qubit* %.q) {
+define void @__quantum__qis__z__body(%Qubit* %.q) {
+  %q = bitcast %Qubit* %.q to %class.QUBIT*
+  call void @quantum__qis__z__body(%class.QUBIT* %q)
+  ret void
+}
+
+define void @__quantum__qis__z__ctl(%Array* %.ctls, %Qubit* %.q) {
   %q = bitcast %Qubit* %.q to %class.QUBIT*
   %ctls = bitcast %Array* %.ctls to %struct.QirArray*
-  call void @quantum__qis__crz(%struct.QirArray* %ctls, double %.theta, %class.QUBIT* %q)
+  call void @quantum__qis__z__ctl(%struct.QirArray* %ctls, %class.QUBIT* %q)
   ret void
 }
+
+;
+; non-standard gates
+;
+define %Result* @__quantum__qis__mz(%Qubit* %.q) {

```

Comment:
Just curious, why is this still needed? When does this particular instruction get included in the code?

Reply:
On its way out...